### PR TITLE
Migrate to vite

### PR DESCRIPTION
**Pull request by Morphenoed**
# :ticket: Description

Updated all dependencies including mui migration and mikroorm update.
Moved from craco to vite.
Improved performance of AllStudentsAdminView

PS. Routes renamed to TutorialRoutes
# :lock: Closes

#1056 #1046

### CODE REVIEW

**Review by nk-coding — CHANGES_REQUESTED** [review]
would it be possible to switch over the github actions to corepack on this branch?
then we could see if the CI runs through
besides that, lgtm

File: .github/workflows/docker-build-test.yml [hunk 1]
```diff
@@ -18,17 +19,24 @@ jobs:
...
-          node-version: 14.x
-      - name: Install pnpm
-        run: npm i -g pnpm@latest-7
+            node-version: '20.x'
+
+      - name: Enable Corepack
+        run: corepack enable
+
+      - name: Prepare pnpm using Corepack
+        run: corepack prepare pnpm@latest --activate
```
Comment (nk-coding) [inline .github/workflows/docker-build-test.yml:32]: IMHO that's a bit dangerous, as this makes builds no longer 100% repeatable
I would recommend setting a pnpm version in the top level package.json, as described [here](https://nodejs.org/api/packages.html#packagemanager)
Also see [here](https://github.com/MiSArch/frontend/blob/main/package.json#L58) for an example

Comment (Morphenoed) [reply .github/workflows/docker-build-test.yml:32]: I've set the version in previous commits already. I've also thought that latest --activate will download the latest available version and thus changed something (newest commits), but then I've noticed that all actions use pnpm@9.6.0 and decided to check. There is no difference between using prepare pnpm@latest --activate or prepare pnpm as both will download the version specified in the package.json. What would you prefer?

Comment (nk-coding) [reply .github/workflows/docker-build-test.yml:32]: I guess the latter one, but your decision

**Review by nk-coding — APPROVED** [review]

**Pull request merged**